pennylane/pauli/dla/lie_closure.py

```diff
@@ -132,13 +133,17 @@ def lie_closure(
 
     epoch = 0
     old_length = 0  # dummy value
-    new_length = len(vspace)
+    new_length = initial_length = len(vspace)
 
     while (new_length > old_length) and (epoch < max_iterations):
         if verbose:
             print(f"epoch {epoch+1} of lie_closure, DLA size is {new_length}")
 
-        for ps1, ps2 in itertools.combinations(vspace.basis, 2):
+        # compute all commutators. We compute the commutators between all newly added operators
+        # and all original generators. This limits the number of commutators added in each
+        # iteration, but it gives us a correspondence between the while loop iteration and the
+        # nesting level of the commutators.
+        for ps1, ps2 in product(vspace.basis[old_length:], vspace.basis[:initial_length]):
```

Comment:
very good to know this built-in!

Reply:
(it is from `itertools`, just to make sure :) )